IBX-8918: Time-slot based models described in User Documentation (#316)
ibexa/documentation-user: docs/personalization/recommendation_models.md

    for requesting recommendations. 
    Recommendation are fetched from all the submodels and merged based on the weight (relevance). 
    If one of the submodels delivers recommendations with better relevance, 
    the results of other models can disappear from the list.

> ### Time-slot based models

Author: Quoting @mnitek-ibexa: _They are quite similar to submodels - they are not stand-alone models, just basic **models** calculated for some limited category._

> Time-slot based models consider only a particular range of time rather than the full day, while creating recommendations.

Reviewer: @dabrt are you sure they are submodels and not models?

They can be used for [popularity](recommendation_models#popularity-models) and [collaborative](recommendation_models#collaborative-models) types of models.

> With time-slot based models, user can configure and set specific time slots.

Author: I'll update with: _you can request to configure (...)_, user doesn't configure it.

These time slots:

- can cover any time frame, including minutes, and don't necessarily have to start and end with full hour
- cannot overlap, for example, you cannot set up 8 a.m - 11 a.m and 9 a.m - 12 a.m slots at once

> - cannot pass between two days, for example, you cannot set slot 11 p.m - 3 a.m

Reviewer 2: ```suggestion
- cannot span between two days, for example, you cannot set a slot to 11 p.m - 3 a.m
```

> In these models, recommendations are created for both configured time slots and for the main model (for example, for the last 30 days).

Reviewer 2: These sentences should go directly after the sentence in line 253

> - can cover any time frame, including minutes (for example, 11 a.m. - 3:30 p.m), and don't necessarily have to start and end at full hour

Reviewer: ```suggestion
- can cover any time frame, including minutes (for example, 11 A.M. - 3:30 P.M.), and don't necessarily have to start and end at full hour
```

> - cannot overlap, for example, you cannot set up 8 a.m - 11 a.m and 9 a.m - 12 a.m slots at once

Reviewer: ```suggestion
- cannot overlap, for example, you cannot set up 8 a.m - 11 A.M and 9 A.M - 12 A.M. slots at once
```

> However, time-slot based recommendations are shown as priority in the hours for which time slots are configured (if requested in recommendation call).

Reviewer 2: ```suggestion
However, time-slot based recommendations are shown as priority in the hours for which time slots are configured (if requested in a recommendation call).
```

> - cannot span between two days, for example, you cannot set slot 11 p.m - 3 a.m

Reviewer: ```suggestion
- cannot span between two days, for example, you cannot set slot 11 P.M. - 3 A.M.
```

Reviewer: According to MS Style Guide.

> - cannot span between two days, for example, you cannot set slot 11 P.M. - 3 A.M.

Reviewer 2: ```suggestion
- cannot span between two days, for example, you cannot set a slot to 11 P.M. - 3 A.M.
```

> To use time slot-based models, this feature must be enabled.

Author: Time-slot based models are described in separate paragraph. 
I don't see a point to move this information to line 191.

> To use time slot-based models, this feature must be enabled.

Reviewer 2: It's your call, however, you added Time-slot models under the "## Submodels" header, which calls for at least mentioning them in an introductory section. "Apart form regular submodels that do this and that you can also apply time slot-based models, which modify the results coming by revommendation or popularity models to provide more hits better aligned with the behavior of users at certain times of the day."

What you could do is rename "Submodels" to, for example, "Modify models" and "Nominal attributes" to "Nominal attribute-based sumbodels", "Numeric attributes" to "Numeric attribute-based sumbodels".
This would bring a little order into this section.

> To use time slot-based models, this feature must be enabled.

Author: I'll just change the level of Time-slot based models: ## instead of ###.

!!! note "Enable time slots"

>     Time slots must be enabled by the administrators.

Author: As discussed with @mnitek-ibexa - updated with 'Ibexa Team' + not needed to explain, how it's done (user doesn't take part in this process).

>     To start using this functionality, first contact the Support Team.

Author: Changing line 15 is out of scope of this PR.
